angular/src/app/project-form/project-form.component.ts

```diff
@@ -66,20 +70,40 @@ export class ProjectFormComponent implements OnInit {
     private projectFormService: ProjectFormService,
     public router: Router,
     private snackBar: MatSnackBar,
+    private tagService: TagsService
     private location: Location
   ) {
   }
 
   ngOnInit() {
     this.project = JSON.parse(localStorage.getItem('project'));
     localStorage.removeItem("project");
-    this.projectForm = this.fb.group({
-      user_id: localStorage.getItem("user_id"),
-      name: [this.project && this.project["name"] || null, Validators.required],
-      description: [this.project && this.project.description || null, Validators.required],
-      url: [this.project && this.project.url || null, Validators.required],
-      github: [this.project && this.project.github || null, Validators.required]
-    });
+
+    // This is kind of a hack because the project doesn't always contain the tags at the moment
+    // so just fetch the tags independently.
+    // When accessing the edit project from from the discover page, the tags are not available,
+    // but when accessing the edit project form from the project detail page they are.
+    if (this.project) {
```

Comment:
The main issue here is the dependency on localStorage. The better way of doing this would be making an API call to fetch the specific project instead of getting the project from localStorage. Currently adding the `withTags=true` param to `api/project/:project_id`, will give you all the project information along with Tags. (e.g. api/project/1?withTags=true will give you a list of tags under `tags` attribute in the JSON object , this is also true for `api/projects?withTags=true,`)
However, this is not the issue in this PR so I will create a separate issue to fix this.